### ISSUE TITLE
Popover with `trigger="hover"` closes when focus is moved to an element inside with tab

### ISSUE THREAD

**Issue by csandman** [chakra-ui/chakra-ui#4665]
### Description

Popover with `trigger="hover"` closes when focus is moved to an element inside with tab, I'd expect it to stay open until focus is moved outside of the context of the popover.

### Link to Reproduction

https://codesandbox.io/s/hover-popover-example-whl09?file=/src/index.js

### Steps to reproduce

I am not sure if this is a bug or feature request, but I'm making a header navigation menu with a popover, and it contains one section of grouped links contained in a popover with `trigger="hover"` set on it.  In order to properly navigate through the header's menu items, I'd need the items in that popover menu to be focusable when tabbing through the links in the header.  I see that focus does properly move to the first item in the popover, however it immediately closes, making it impossible to focus on them using keyboard navigation.

To see what I'm talking about in the demo, open the codesandbox and click on the output screen, and tab through the elements.

If anyone has a suggestion or workaround for this I'd be happy to hear it!

### Chakra UI Version

1.6.7

### Browser

Google Chrome

### Operating System

- [X] macOS
- [ ] Windows
- [ ] Linux

**Comment by takethefake** [comment]
Hej @csandman thanks for reaching out.

I've checked your sandbox and was wondering if the usage of `Menu` (https://chakra-ui.com/docs/overlay/menu) might solve your problem?

You can find a working example of your use case with `Menu` here: https://codesandbox.io/s/hover-popover-example-forked-7yxo7

**Comment by csandman** [comment]
@takethefake  With that menu example it still appears that it does not open on hover which is the main initial interaction I'm looking for

**Comment by csandman** [comment]
In case anyone else runs into this issue, I was able to manually make this work by following a couple steps. Here is a demo:

https://codesandbox.io/s/hover-popover-example-solution-64z6v?file=/src/index.js

And here is what I did to make it work:

1. Make the popover manually controlled, but keep its trigger set to `hover`
2. Add the following event listeners to both the child of the `PopoverTrigger` and to the `PopoverContent` itself:

```js
onMouseEnter={onOpen}
onMouseLeave={onClose}
onFocus={onOpen}
onBlur={onClose}
```

3. Add a pseudo element to the `PopoverContent` to fill the gap between the popover and the trigger:

```js
_after={{
  content: "''",
  position: "absolute",
  w: "100%",
  h: `${gutter}px`,
  top: `-${gutter}px`,
  left: 0
}}
```
